music_assistant/server/providers/bluesound/__init__.py

```diff
@@ -0,0 +1,586 @@
+    # The ConfigEntry is a dataclass that represents a single configuration entry.
+    # The ConfigValueType is an Enum that represents the type of value that
+    # can be stored in a ConfigEntry.
+    # If your provider does not need any configuration, you can return an empty tuple.
+    # ruff: noqa: ARG001
+    return ()
+
+
+class BluesoundDiscoveryInfo(TypedDict):
+    """Template for MDNS discovery info."""
+
+    _objectType: str
+    ip_address: str
+    port: str
+    mac: str
+    model: str
+    zs: bool
+
+
+class BluesoundPlayer:
+    """Holds the details of the (discovered) BluOS player."""
+
+    def __init__(
+        self,
+        prov: BluesoundPlayerProvider,
+        player_id: str,
+        discovery_info: BluesoundDiscoveryInfo,
+        ip_address: str,
+        port: int,
+    ) -> None:
+        """Initialize the BluOS Player."""
+        self.port = port
+        self.prov = prov
+        self.mass = prov.mass
+        self.player_id = player_id
+        self.discovery_info = discovery_info
+        self.ip_address = ip_address
+        self.logger = prov.logger.getChild(player_id)
+        self.connected: bool = True
+        self.client = BluosPlayer(self.ip_address, self.port, self.mass.http_session)
+        self.sync_status = SyncStatus
+        self.status = Status
+        self.mass_player: Player | None = None
+        self._listen_task: asyncio.Task | None = None
+
+    async def disconnect(self) -> None:
+        """Disconnect the client and cleanup."""
+        if self._listen_task and not self._listen_task.done():
+            self._listen_task.cancel()
+        if self.client:
+            await self.client.close()
+        self.connected = False
+        self.logger.debug("Disconnected from player API")
+
+    async def update_attributes(self) -> None:
+        """Update the player attributes."""
+        # self.logger.debug("Update attributes")
+
+        self.sync_status = await self.client.sync_status()
+        self.status = await self.client.status()
+
+        if not self.mass_player:
+            return
+        if self.sync_status.volume == -1:
+            self.mass_player.volume_level = 100
+        else:
+            self.mass_player.volume_level = self.sync_status.volume
+        self.mass_player.volume_muted = self.status.mute
+
+        # self.logger.debug(self.status.input_id)
+        if self.status.state == "stream" and self.status.input_id == "input0":
+            self.mass_player.active_source = SOURCE_LINE_IN
+        elif self.status.state == "stream" and self.status.input_id == "Airplay":
+            self.mass_player.active_source = SOURCE_AIRPLAY
+        elif self.status.state == "stream" and self.status.input_id == "Spotify":
+            self.mass_player.active_source = SOURCE_SPOTIFY
+        elif self.status.state == "stream" and self.status.input_id == "RadioParadise":
+            self.mass_player.active_source = SOURCE_RADIO
+
```

Comment:
If you have issues finding out if MA is the active source - you could also just optimistically set it from the play-media